File: eBarbershop.Model/Narudzba.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eBarbershop.Model
{
    public class Narudzba
    {
        public int NarudzbaId { get; set; }
        public DateTime Datum { get; set; }
        public decimal UkupnaCijena { get; set; }

        public int KorisnikId { get; set; }

        public ICollection<NarudzbaProizvodi> NarudzbaProizvodis { get; set; }
    }

}

File: UI/ebarbershop_admin/lib/models/narudzba.dart
import 'package:ebarbershop_admin/models/narudzbaproizvodi.dart';
import 'package:json_annotation/json_annotation.dart';

part 'narudzba.g.dart';

@JsonSerializable()
class Narudzba {
  int? narudzbaId;
  DateTime? datum;
  int? korisnikId;
  double? ukupnaCijena;
  List<NarudzbaProizvodi>? narudzbaProizvodis;

  Narudzba(this.narudzbaId, this.datum, this.korisnikId, this.ukupnaCijena,
      this.narudzbaProizvodis);

  factory Narudzba.fromJson(Map<String, dynamic> json) =>
      _$NarudzbaFromJson(json);
  Map<String, dynamic> toJson() => _$NarudzbaToJson(this);
}
  
